tests/io/test_io.py

```diff
@@ -281,6 +282,11 @@ def test_import_file_io_does_not_exist() -> None:
     assert _import_file_io("pyiceberg.does.not.exist.FileIO", {}) is None
 
 
+def test_import_file_io_logs_exception(caplog: Any) -> None:
+    _import_file_io("pyiceberg.does.not.exist.FileIO", {})
+    assert "ModuleNotFoundError: No module named 'pyiceberg.does'" in caplog.text
```

Comment:
does the error just return `pyiceberg.does` or the entire import, `pyiceberg.does.not.exist.FileIO`

Reply:
It fails here on `pyiceberg.does`. I presume that's Python trying and failing to import higher-level modules first?

Reply:
thanks! i was curious about the behavior

Reply:
Yeah, was also surprised